internal/provider/resource_bgp_session.go

```diff
@@ -471,16 +481,31 @@ func extractBgpSessionUpdate(d *schema.ResourceData) packetfabric.BgpSession {
 	// Azure BGP session Update: l3_address = Azure Subnet (primary or secondary)
 	// set l3Address based on the values of primarySubnet and secondarySubnet when modified
 	// This is a temporary solution until the BGP API is refactored.
-	if d.HasChange("primary_subnet") {
-		if primarySubnet, ok := d.GetOk("primary_subnet"); ok {
-			bgpSession.L3Address = primarySubnet.(string)
+	var primary, secondary string
+	if primarySubnet, ok := d.GetOk("primary_subnet"); ok {
+		primary = primarySubnet.(string)
+		if d.HasChange("primary_subnet") {
+			bgpSession.L3Address = primary
```

Comment:
Why assign the primary subnet to the l3_address directly? If there is a change, then subnet +1 should be l3_address and subnet +2 should be the remote_address

Reply:
Good point, I can back that out... I just switched the loops because I thought I would need the primarySubnet but then I didn't and forgot to put it back